addons/blocks2image/userscript.js

```diff
@@ -0,0 +1,260 @@
+  function eventMouseDown(e) {
+    if (e.button === 2) {
+      let blockSvg = e.target.closest("[data-id]");
+      let isBackground = !blockSvg && e.target.closest("svg.blocklySvg");
+      if (blockSvg || isBackground) {
+        let dataId = blockSvg && blockSvg.getAttribute("data-id");
+        if (dataId || isBackground) {
+          setTimeout(async () => {
+            let widget = document.querySelector("div.blocklyWidgetDiv");
+            if (!widget) {
+              return;
+            }
+            let blocklyContextMenu = widget.querySelector("div.blocklyContextMenu");
+            if (!blocklyContextMenu) {
+              return;
+            }
+            if (isBackground) {
+              buttonItem.forEach((item, index) => {
+                const wrapperItem = document.createElement("div");
+                wrapperItem.id = `blocks2imgCommand${index + 1}`;
+                wrapperItem.classList.add("goog-menuitem", "blocks2img");
+                wrapperItem.onmouseenter = () => wrapperItem.classList.add("goog-menuitem-highlight");
+                wrapperItem.onmouseleave = () => wrapperItem.classList.remove("goog-menuitem-highlight");
+                wrapperItem.style.userSelect = "none";
+                if (index === 0) {
+                  wrapperItem.style.borderTop = "1px solid hsla(0, 0%, 0%, 0.15)";
+                }
+                const menuItem = document.createElement("div");
+                menuItem.classList.add("goog-menuitem-content");
+                menuItem.style.userSelect = "user-select: none";
+                menuItem.textContent = item;
+                wrapperItem.append(menuItem);
+                wrapperItem.onclick = () => {
+                  hidePopups();
+                  exportBlock({ command: "export" + (index + 1) });
+                };
+                blocklyContextMenu.append(wrapperItem);
+              });
+            }
+
+            if (blocklyContextMenu.children.length < 15) {
+              blocklyContextMenu.style.maxHeight = "none";
+              widget.style.height = blocklyContextMenu.getBoundingClientRect().height + 12 + "px";
+              blocklyContextMenu.style.maxHeight = "";
+            }
+
+            function hidePopups() {
+              const currentWorkspace = Blockly.getMainWorkspace();
+              const element = currentWorkspace.getToolbox().HtmlDiv;
+              element.dispatchEvent(new MouseEvent("mousedown", { relatedTarget: element, bubbles: true }));
+              element.dispatchEvent(new MouseEvent("mouseup", { relatedTarget: element, bubbles: true }));
+            }
+          }, 1);
+        }
+      }
+    }
+  }
+
+  while (true) {
+    let blocklyWorkspace = await addon.tab.waitForElement("g.blocklyWorkspace", {
+      markAsSeen: true,
+    });
+    if (!document.querySelector("[class^='author-info_username-line']")) {
+      // insert contextmenu
+      blocklyWorkspace.addEventListener("mousedown", (e) => eventMouseDown(e));
```

Comment:
@summerscar Why is the event not registered if that element does not exist? During testing, the context menu buttons would not appear because of this. Can you explain the motivation for this?

Reply:
I think this is redundant code, this is  before  copy form  other addon which i refer to, now  I have deleted

Reply:
This all will probably be changed when we migrate to using the new blockly context menu API... It will simplify things a lot...

Reply:
@thecolaber bruh not related at all